### PR TITLE
Documented the detectCorrupted and corruptedMessage options

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | master (3.1+) |
| Fixed tickets | https://github.com/symfony/symfony/pull/17458 |

### CODE REVIEW

File: reference/constraints/Image.rst [hunk 1]
```diff
@@ -311,6 +313,18 @@ allowPortrait
 
 If this option is false, the image cannot be portrait oriented.
 
+detectCorrupted
+~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.1
+     The ``detectCorrupted`` option was introduced in Symfony 3.1.
+
+**type**: ``Boolean`` **default**: ``false``
```
Comment (xabbuh) [inline reference/constraints/Image.rst]: boolean

File: reference/constraints/Image.rst [hunk 2]
```diff
@@ -311,6 +313,18 @@ allowPortrait
...
+
+If this option is true, the image contents are validated to ensure that the
+image is not corrupted. This validation is done with PHP's ``imagecreatefromstring()``
```
Comment (xabbuh) [inline reference/constraints/Image.rst]: Let's use the `phpfunction` role here.

File: reference/constraints/Image.rst [hunk 3]
```diff
@@ -392,4 +406,15 @@ Portrait oriented images are not allowed``
...
+corruptedMessage
+~~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.1
+     The ``corruptedMessage`` option was introduced in Symfony 3.1.
+
+**type**: ``string`` **default**: ``The image file is corrupted.``
+
+The error message if the image contents are corrupted and you set `detectCorrupted`_
+to ``true``.
```
Comment (xabbuh) [inline reference/constraints/Image.rst]: I would change the order:

> [...] when the `detectCorrupted` options is enabled and the image is corrupted.

File: reference/constraints/Image.rst [hunk 4]
```diff
@@ -311,6 +313,18 @@ allowPortrait
...
+~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.1
+     The ``detectCorrupted`` option was introduced in Symfony 3.1.
+
+**type**: ``Boolean`` **default**: ``false``
+
+If this option is true, the image contents are validated to ensure that the
+image is not corrupted. This validation is done with PHP's ``imagecreatefromstring()``
+function, which requires the PHP GD extension to be enabled.
```
Comment (OskarStark) [inline reference/constraints/Image.rst]: should we link to the GD Extension?

**Comment by javiereguiluz** [comment]
Thanks for the review! I've done all the changes. This is now ready to be merged. Thanks!

**Comment by xabbuh** [comment]
👍

**Comment by WouterJ** [comment]
:+1: Thanks!

File: reference/constraints/Image.rst [hunk 5]
```diff
@@ -392,4 +406,16 @@ Portrait oriented images are not allowed``
...
 
+corruptedMessage
+~~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.1
+     The ``corruptedMessage`` option was introduced in Symfony 3.1.
+
+**type**: ``string`` **default**: ``The image file is corrupted.``
+
+The error message when the `detectCorrupted`_ option is enabled and the image
```
Comment (WouterJ) [inline reference/constraints/Image.rst]: I was to unsure to commit this, but shouldn't we use `if` here instead of `when`?

**Pull request closed without merging**